packages/app/src/elements/field.ts

```diff
@@ -200,6 +228,123 @@ export class FieldElement extends BaseElement {
         }
     }
 
+    /**
+     * Checks against HIBP's API for the number of times the user's password has been detected in a breach.
+     * @private
+     */
+    private async _checkPasswordBreach() {
+        if (!this.value) {
+            this._passwordBreachCount = -1;
+            return;
+        }
+
+        const sha1Hash = crypto.createHash("sha1")
+            .update(this.value)
+            .digest("hex");
```

Comment:
Again, please use the `@padloc/core/src/crypto` module for all cryptographic needs.